### PR TITLE
Refactor command line handling of namespaces

### PR DESCRIPTION
**Pull request by pauloue**
https://github.com/luarocks/luarocks/issues/1067

Rock name and namespace are always passed around separately to avoid splitting them multiple times.

### CODE REVIEW

File: src/luarocks/cmd/write_rockspec.lua [hunk 1]
```diff
@@ -67,6 +67,7 @@ rockspec, and is not guaranteed to be complete or correct. ]], util.see_also())
 
    cmd:argument("name", "Name of the rock.")
       :args("?")
+      :action(util.namespaced_name_action)
```
Comment (pauloue) [inline src/luarocks/cmd/write_rockspec.lua]: Does write_rockspec even need to accept a namespace?

File: src/luarocks/util.lua [hunk 2]
```diff
@@ -373,6 +353,21 @@ function util.split_namespace(ns_name)
    return ns_name
 end
 
+--- Argparse action callback for namespaced rock arguments.
+function util.namespaced_name_action(args, target, ns_name)
```
Comment (daurnimator) [inline src/luarocks/util.lua]: This looks a lot like `adjust_name_and_namespace`: could you modify the behaviour of that function?
Also e.g. keep the `assert`s?

File: src/luarocks/cmd/write_rockspec.lua (hunk 1 above)
Comment (hishamhm) [reply src/luarocks/cmd/write_rockspec.lua]: I don't think so, I think that was just blanket-added to every command that used a `name`...

**Comment by hishamhm** [comment]
@pauloue I fixed the conflicts and merged manually. Thank you for this!

**Pull request closed without merging**